esp/public/media/scripts/onsite/teacher_checkin.js

```diff
@@ -108,28 +118,54 @@ $j(function(){
         this.value += "...";
     });
 
+    // Undo the check in of the user with the specified username, assuming
+    // that that user was not checked in when this page was loaded (because
+    // the DOM would look different depending on whether that were true). If
+    // no username specified, undo the last check in.
+    function undoLiveCheckIn(username) {
+        var targetCheckin;
+        if (username === undefined) {
+            if (checkins.length === 0) return false;
+            targetCheckin = checkins[checkins.length - 1];
+        } else {
+            // search from the back because we assume more recent checkins
+            // are more likely to be undone
+            for (var i = checkins.length - 1; i >= 0; i--) {
+                if (checkins[i].username === username) {
+                    targetCheckin = checkins[i];
+                    break;
+                }
+            }
+        }
+        if (targetCheckin === undefined) return;
+        username = targetCheckin.username;
+        var $td = targetCheckin.$td;
+        var $msg = $td.children('.message');
+        undoCheckIn(username, function(response) {
+            // find targetCheckin in checkins and splice it out
+            for (var i = checkins.length - 1; i >= 0; i--) {
+                if (checkins[i] === targetCheckin) {
+                    checkins.splice(i, 1);
+                    break;
+                }
+            }
+            $msg.html(response.message+"<br/>");
+            $td.children('.checkin').show().prop('disabled', false);
+            $td.prev().prop('class', 'not-checked-in');
+            selected = $j('.checkin:enabled').index($td.find('.checkin'));
+            updateSelected(true);
+        });
+        $msg.append(' Undoing...');
```

Comment:
Yeah, it'll go after the Undo button and we want some space to visually separate them.